src/Controller/CheckoutRequestController.php

```diff
@@ -252,4 +254,25 @@ protected function processException($errorType, $errorMessage, \Exception $excep
 
         return $this->getResponse()->withJson($errorResp)->withStatus($statusCode);
     }
+
+    /**
+     * @param array $data should have a patronBarcode attribute, which will be randomly reassigned for partners
+     * @return array will be a copy of the input with patronBarcode reassigned in case it was originally a partner barcode
+     */
+
+    static function reassignPartnerBarcode($data)
```

Comment:
We try to avoid mixing static and non-static functions unless there's a case for this being called independent of the class itself. If so, use `public static function` instead. But it looks like a method unique to this class so use `public function` and change the call above from `self::randomizePartnerBarcodes` to `$this->randomizePartnerBarcodes`

Reply:
Fixed